enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ConfigFileProvider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.cloud.oracle.assets;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.netbeans.spi.lsp.CommandProvider;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Dusan Petrovic
+ */
+@ServiceProvider(service = CommandProvider.class)
+public class ConfigFileProvider implements CommandProvider {
+    
+    private static final String GET_CONFIG_FILE_PATH = "nbls.config.file.path"; //NOI18N
+       
+    private final ConfigFileGenerator configFileGenerator;
+    
+    public ConfigFileProvider() {
+        this.configFileGenerator = new ConfigFileGenerator("app-config-", ".properties", GET_CONFIG_FILE_PATH, false); // NOI18N
+    }
+    
+    @Override
+    public Set<String> getCommands() {
+        return Collections.singleton(GET_CONFIG_FILE_PATH);
+    }
+
+    @Override
+    public CompletableFuture<Object> runCommand(String command, List<Object> arguments) {
+        CompletableFuture ret = new CompletableFuture();
+
+        Properties props = new Properties();
+        PropertiesGenerator propGen = new PropertiesGenerator(false);
+        props.putAll(propGen.getApplication());
+        props.putAll(propGen.getBootstrap());
+        
+        if (!props.isEmpty()) {
+            Path temp;
+            
+            try {
+                temp = configFileGenerator.generateConfigFile();
```

Comment:
Instead of two calls (`generateConfigFile`, `writePropertiesToFile`) that happen in the sequence with double error handling, could there be just a `writePropertiesFile` that makes the directory, creates temp file and populates it in one call - centralizing IOException handling (delete the file being created) inside ConfigFileGenerator ?